enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+to result in only the manifests annotated for the specified cluster
+profile to be processed by the resulting cluster. Developer tools team
+members have proposed CVO patches as part of that enhancement, and
+will continue to participate in the review process, but progress
+getting the implementation merged has been slow so far. Having this
+enhancement as an additional use case for the cluster profile
+mechanism hopefully will help clarify how that feature is being used
+and expedite its implementation.
+
+Another risk is that it will take too long to get the initial
+annotations merged to all the needed OpenShift component repositories
+during the first phase of development. Until all needed annotations
+are merged, the SNC tooling cannot be updated to make use of the
+single-node-developer cluster profile, as needed components would be
+excluded from the cluster. The review process for this enhancement
+should help raise awareness across the various teams that will need to
+review and merge these patches.
+
+A final risk is that new components will be added to OpenShift after
+the completion of phase 1 that should be included in single node
+developer clusters, but will not have the necessary annotation
+included in their manifests due to lack of awareness or concern for
+the CRC product. Given that single node developer clusters are
+expected to pass the majority of the OpenShift conformance tests,
+component developers should include this profile's annotation unless
+the new component is clearly not relevant to the development of
+applications on single node clusters. Developer tools team members
+will be available to collaborate with component developers regarding
+whether a component should be included in the single-node-developer
+cluster profile, and if so, how it might be configured to minimize
+resource requirements.
+
+## Design Details
+
+### Open Questions
+
+One currently open question is how to manage the set of operators
+visible within the operator hub on a single-node-developer
+cluster. Operators that cannot run without OpenShift components
+excluded from the profile should not be available to install. Its not
+clear whether this is best achieved by producing a custom index for
+the OLM, having the OLM honor annotations in the manifests of
+operators it manages, or by dynamically evaluating whether
+dependencies are available in the cluster.
+
+### Test Plan
+
+Individual OpenShift components should not require any additional test
+cases during phase one, as the annotation for the single node
+developer cluster will be applied to existing manifest files that are
+already tested. Once these annotations are all merged and the SNC
+tooling is updated to have the OpenShift installer use the new cluster
+profile, the existing SNC tests and CRC product tests will verify that
+nothing has changed in the resulting cluster.
+
+Once phase one is complete, a new OpenShift CI job will be added to
```

Comment:
will it be a blocking CI job for every repo?

Reply:
Looking for guidance on this.

Reply:
periodic until it is demonstrated to work reliably.